src/items/powerup_manager.cpp

```diff
@@ -205,142 +208,343 @@ void PowerupManager::LoadPowerup(PowerupType type, const XMLNode &node)
 }   // LoadNode
 
 // ----------------------------------------------------------------------------
-/** Loads a weight list specified in powerup.xml. The different position
- *  classes must be loaded in the right order
+/** Loads a weight list specified in powerup.xml.
+ *  Calculates a linear average of the weight specified for the reference points
+ *  of num_karts higher and lower (unless it exactly matches one)
  *  \param root The root node of powerup.xml
+ *  \param num_karts The number of karts in the race
  *  \param class_name The name of the position class to load.
  *  \param position_class The class for which the weights are read.
  */
 void PowerupManager::loadWeights(const XMLNode &root,
+                                 unsigned int num_karts,
                                  const std::string &class_name,
                                  PositionClass position_class)
 {
-    const XMLNode *node = root.getNode(class_name);
-    std::string s(""), s_multi("");
-    if(node) node->get("w",       &s      );
-    if(node) node->get("w-multi", &s_multi);
+    const XMLNode *node = root.getNode(class_name), *node2 = root.getNode("reference_points");
+    std::string s_inf(""), s_inf_multi(""), s_sup(""), s_sup_multi(""), ref;
```

Comment:
All those variables don't have very clear names. Choosing better names, or adding documentation comments, could help clarify that a lot